test/e2e/basic_provisioning_test.go

```diff
@@ -118,6 +123,143 @@ var _ = Describe("Provisioning", func() {
+		time.Sleep(2 * time.Second)
+
+		err = clusterProxy.GetClient().Delete(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the deleting state")
+		Eventually(func() (string, error) {
+			var currentBmh metal3api.BareMetalHost
+			err := clusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: bmh.Name, Namespace: bmh.Namespace}, &currentBmh)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// If the BMH is not found, we assume it has been successfully deleted.
+					return "deleting", nil
+				}
+				// Any other error should be returned.
+				return "", err
+			}
+
+			currentStatus := currentBmh.Status.Provisioning.State
+
+			// If the state is 'deleting', we return this state to finish the polling.
+			if currentStatus == "deleting" {
+				return "deleting", nil
+			}
+
+			// If the state is empty or 'provisioned', we continue polling.
+			if currentStatus == "" || currentStatus == "provisioned" {
+				return "", nil
+			}
+
+			// Any other state is unexpected, and we stop the polling.
+			return "", StopTrying(fmt.Sprintf("BMH is in an unexpected state: %s", currentStatus))
+		}, e2eConfig.GetIntervals(specName, "wait-deleting")...).Should(Equal("deleting"))
+
+		By("Waiting for the secret to be deleted")
+		Eventually(func() bool {
+			err := clusterProxy.GetClient().Get(ctx, client.ObjectKey{
+				Name:      "bmc-credentials",
+				Namespace: namespace.Name,
+			}, &corev1.Secret{})
+			return apierrors.IsNotFound(err)
+		}, e2eConfig.GetIntervals(specName, "wait-secret-deletion")...).Should(BeTrue())
+
+		By("Creating a secret with BMH credentials")
+		secretName := "bmc-credentials"
+		bmcCredentials = corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: namespace.Name,
+			},
+			StringData: map[string]string{
+				"username": bmcUser,
+				"password": bmcPassword,
+			},
+		}
+
+		err = clusterProxy.GetClient().Create(ctx, &bmcCredentials)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Recreating the BMH with the previously saved status in the status annotation")
+		bmh = metal3api.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      specName,
+				Namespace: namespace.Name,
+				Annotations: map[string]string{
+					metal3api.StatusAnnotation: string(statusJSON),
+				},
+			},
+			Spec: metal3api.BareMetalHostSpec{
+				Online: true,
+				BMC: metal3api.BMCDetails{
+					Address:         bmcAddress,
+					CredentialsName: "bmc-credentials",
+				},
+				BootMode:       metal3api.Legacy,
+				BootMACAddress: bootMacAddress,
+				Image: &metal3api.Image{
+					URL:      e2eConfig.GetVariable("IMAGE_URL"),
+					Checksum: e2eConfig.GetVariable("IMAGE_CHECKSUM"),
+				},
+				RootDeviceHints: &metal3api.RootDeviceHints{
+					DeviceName: "/dev/vda",
+				},
+			},
+		}
+
+		err = clusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the 'provisioned' state")
```

Comment:
nit: Maybe word this a bit differently so it is obvious that it should skip inspection and provisioning
```suggestion
		By("Checking that the BMH goes directly to 'provisioned' state")
```